c++/README.md

```diff
@@ -0,0 +1,22 @@
+# C++
+
+C++ and C code should conform to the [Google coding conventions](https://google.github.io/styleguide/cppguide.html) in terms of variable names and layout.
```

Comment:
Plz can has `#pragma once`?

- It's almost standard, even if it's not strictly compliant with Google style guide. Almost all modern compilers support it, and if we're C++11 or greater, we're probably safe.
- It's also explicitly listed as an acceptable alternative in PHAS0100 (in the assignments, at least).
- It's a couple less lines of code per file, which makes the mathematician in me happy. It also means I don't have to `ifndef` my entire header file.

I am willing to be downvoted on this :sweat_smile:

Reply:
Bbut .... what about that one exotic architecture that nobody uses?

Reply:
Yeah I'd agree that `#pragma once` is usually fine for application projects, although if we mention it explicitly we should make it clear that it's technically non-standard. I would probably think twice about using it in a library project that was intended for wide use just to be on the safe side.

Reply:
It's recommended against in Google and [ISO](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#sf8-use-include-guards-for-all-header-files) styles.

* https://github.com/google/styleguide/issues/538

 Fwiw, I like `#pragma once`. It's 1 line instead of 3. But do we want to add a list of our own exceptions?

Reply:
We did for TDMS's developer standard/docs...

IMO, a single exception is not a list. And it's not a controversial exception either. Maybe flavour it with something like

```
We prefer the use of `#pragma once` over `ifndef`-style include guards where appropriate; for example in application-focused projects. For development of libraries or codebases that will see a wide usage across many different architectures, project teams should discuss whether use of `#pragma` is suitable based on the context.
```

Reply:
Oh, I definitely don't think we should _prefer_ `#pragma once`. I think it's _acceptable_ in most projects, but if we need to pick one to express a preference, then the winner is definitely traditional include guards because they're fully compliant, and avoid some issues with the compiler having to decide when something is the same file or not. I generally just use `#pragma once` for small projects that I know where they're going to be run.